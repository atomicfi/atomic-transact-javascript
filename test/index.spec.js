const { Atomic, Product } = require('../index')
const { pick } = require('lodash')

describe('JavaScript SDK', () => {
  const mocks = {}
  let iframeElement

  beforeEach(() => {
    navigator = {
      platform: 'ios',
      vendor: 'mac'
    }

    mocks.documentAppendChild = jest.fn()
    mocks.elementSetAttribute = jest.fn()
    iframeElement = {
      setAttribute: mocks.elementSetAttribute,
      style: {}
    }
    mocks.documentCreateElement = jest.fn().mockReturnValue(iframeElement)
    mocks.windowAddEventListener = jest.fn()

    document = {
      body: { style: {}, appendChild: mocks.documentAppendChild },
      createElement: mocks.documentCreateElement
    }
    window = {
      addEventListener: mocks.windowAddEventListener
    }
  })

  it('should initialize the sdk', () => {
    Atomic.transact({
      config: {
        publicToken: '6e93549e-3571-4f57-b0f7-77b7cb0b5e48',
        tasks: [{ product: Product.DEPOSIT }]
      }
    })

    expect(mocks.documentAppendChild).toHaveBeenCalled()
    expect(iframeElement.setAttribute.mock.calls).toMatchSnapshot()
    expect(pick(iframeElement, ['style', 'src', 'id'])).toMatchSnapshot()
  })

  it('should initialize inside of a container', () => {
    // TODO
  })

  it('should responsd to Transact events', () => {
    // TODO
  })

  it('should be able to manually close the session', () => {
    // TODO
  })
})
