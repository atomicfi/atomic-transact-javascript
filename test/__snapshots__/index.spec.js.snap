// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JavaScript SDK should initialize the sdk 1`] = `
[
  [
    "allow",
    "web-share",
  ],
  [
    "title",
    "Atomic Interface",
  ],
  [
    "aria-label",
    "Atomic Interface",
  ],
]
`;

exports[`JavaScript SDK should initialize the sdk 2`] = `
{
  "id": "atomic-transact-iframe",
  "src": "https://transact.atomicfi.com/initialize/eyJpblNkayI6dHJ1ZSwicGxhdGZvcm0iOnsibmFtZSI6ImJyb3dzZXIiLCJzZGtWZXJzaW9uIjoiMy4wLjQiLCJzeXN0ZW1WZXJzaW9uIjoiaW9zLW1hYyJ9LCJwdWJsaWNUb2tlbiI6IjZlOTM1NDllLTM1NzEtNGY1Ny1iMGY3LTc3YjdjYjBiNWU0OCIsInRhc2tzIjpbeyJwcm9kdWN0IjoiZGVwb3NpdCJ9XX0=",
  "style": {
    "cssText": "width: 100%; height: 100%; z-index: 888888888; border-width: 0; overflow-x: hidden; overflow-y: auto; position: fixed; top: 0; left: 0; right: 0; bottom: 0;",
  },
}
`;
